catkit/hardware/iris_ao/iris_ao_controller.py

```diff
@@ -69,6 +69,32 @@ def initialize(self,
         # Where to write ConfigPTT.ini file that gets read by the C++ code
         self.filename_ptt_dm = filename_ptt_dm
 
+    def _communicate(self, input):
+        # Poll in case it died.
+        if self.instrument.poll() is not None:
+            self.instrument = None  # If it died, there's nothing to close, so don't.
+            raise RuntimeError(f"{self.config_id} unexpectedly closed: ('{self.instrument.returncode}').")
+
+        # Write.
+        try:
+            self.instrument.stdin.write(input)
+            self.instrument.stdin.flush()
+        except Exception as error:
+            self.instrument = None  # If it died, there's nothing to close, so don't.
+            raise RuntimeError(f"{self.config_id} unexpectedly closed.") from error
+
+        # Poll in case it died (though the above flush would most likely catch this).
+        #if self.instrument.poll() is not None:
+        #    self.instrument = None  # If it died, there's nothing to close, so don't.
+        #    raise RuntimeError(f"{self.config_id} unexpectedly closed: ('{self.instrument.returncode}').")
+
+        # Read response.
+        self.log.info(f"{self.config_id}: Waiting for response...")
+        resp = self.instrument.stdout.readline().decode()
```

Comment:
This could be an issue as it could block indefinitely.

Reply:
@ivalaginja If the driver code blocks when talking to the dm such that it doesn't return until it's actually ready, this read will also block and wait until the driver func call returns, then our C++ writes to ``stdout``, which we read thus syncing Python and the DM.

If it was a sync issue of this nature that you were having with iterative dm communications, this should fix that problem. But that's quite a few ifs.

Reply:
I've added a hardware test for this [here](https://github.com/spacetelescope/hicat-package/blob/119a0e072fc4eebb2436ee809391cdd1b8ed6943/hicat/hardware/iris_ao/tests/hardware_speed.py).

Reply:
Though, that test only pokes single segments which may be faster than having to do all of them per iter.

Reply:
Many ifs, but might rule some things out. Excellent that you put this in place, thanks!
I don't think doing a single segment is much faster. The IrisAO is both extremely repeatable and extremely fast in its commands, and on JOST we observed the deadness of connection in iterations that did something that's almost exactly what you wrote up in the test, just tilting a single segment more and more, iteratively. We'll  have to try and see.

Reply:
See https://github.com/spacetelescope/hicat-package/pull/490#issuecomment-772775083, this PR solves your problem as I can reproduce your stall without the added ``readline()`` wait.